### PR TITLE
[DX-342] update menu generator script to generate show status in menu.yaml

### PR DESCRIPTION
**Pull request by dcs3spp**
update menu generator script to generate show status in menu.yaml

## Types of changes
- [ ] Fixing typo (please merge to production)
- [ ] Documenting a new feature (please merge to production)
- [ ] Documentation for future release (please do not merge to production)
- [x] Something else (please add if needs merging to production or not)

## Checklist

- [x] I have [reviewed the guidelines](../CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](../CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I have added a preview link.

### CODE REVIEW

File: tyk-docs/data/menu.yaml [hunk 1]
```diff
@@ -1,2344 +1,3170 @@
...
   - title: "Deployment and Operations"
     path: /apim
+    show: true
     category: Tab
     menu:
     - title: "Overview"
+      show: false
       category: Page
     - title: "Quick start guides"
+      show: true
```
Comment (yurisasuke) [inline tyk-docs/data/menu.yaml]: This should be false since it does not have any child.

File: tyk-docs/data/menu.yaml [hunk 2]
```diff
@@ -1,2344 +1,3170 @@
...
         - title: "Securing System Payloads"
           path: /tyk-configuration-reference/securing-system-payloads
+          show: true
           category: Page
       - title: "API Documentation"
+        show: true
         category: Directory
         menu:
         - title: "Overview"
+          show: true
```
Comment (yurisasuke) [inline tyk-docs/data/menu.yaml]: This should be false as the menu is empty (it has no child)

**Review by yurisasuke — COMMENTED** [review]
When an item is in the category "Directory"  and all its children's show fields are false, we should also set its show field to false.

**Review by yurisasuke — APPROVED** [review]

**Comment by dcs3spp** [comment]
Converted to draft PR since we are using the page path and menu path to determine show status

**Comment by dcs3spp** [comment]
Closing since issue addressed by https://github.com/TykTechnologies/tyk-docs/pull/2691/

**Pull request closed without merging**